common/src/services/crypto.service.ts

```diff
@@ -336,74 +325,71 @@ export class CryptoService implements CryptoServiceAbstraction {
     }
 
     async hasKey(): Promise<boolean> {
-        return this.hasKeyInMemory() || await this.hasKeyStored('auto') || await this.hasKeyStored('biometric');
+        return await this.hasKeyInMemory() || await this.hasKeyStored(KeySuffixOptions.Auto) || await this.hasKeyStored(KeySuffixOptions.Biometric);
     }
 
-    hasKeyInMemory(): boolean {
-        return this.key != null;
+    async hasKeyInMemory(userId?: string): Promise<boolean> {
+        return await this.stateService.getCryptoMasterKey({ userId: userId }) != null;
     }
 
-    hasKeyStored(keySuffix: KeySuffixOptions): Promise<boolean> {
-        return this.secureStorageService.has(Keys.key, { keySuffix: keySuffix });
+    async hasKeyStored(keySuffix: KeySuffixOptions, userId?: string): Promise<boolean> {
+        return await this.stateService.getCryptoMasterKeyB64({ keySuffix: keySuffix, userId: userId }) != null;
     }
 
     async hasEncKey(): Promise<boolean> {
-        const encKey = await this.storageService.get<string>(Keys.encKey);
-        return encKey != null;
+        return await this.stateService.getEncryptedCryptoSymmetricKey() != null;
     }
 
-    async clearKey(clearSecretStorage: boolean = true): Promise<any> {
-        this.key = this.legacyEtmKey = null;
+    async clearKey(clearSecretStorage: boolean = true, userId?: string): Promise<any> {
+        await this.stateService.setCryptoMasterKey(null, { userId: userId });
```

Comment:
Dow e also need to clear the B64 key?

Reply:
Yes, that's what happens in `clearSecretKeyStore()` if `clearSecretStorage` is true.